torch/csrc/profiler/collection.cpp

```diff
@@ -38,22 +38,12 @@ void InputOutputEncoder::push(c10::ArrayRef<const c10::IValue> values) {
 void InputOutputEncoder::push(const at::Tensor& t) {
   if (t.defined()) {
     tags_.emplace_back(Tag::Tensor);
-    const auto& sizes = t.sizes();
-    const auto dim = sizes.size();
-    TORCH_CHECK(
-        dim <= std::numeric_limits<uint32_t>::max(),
-        "Cannot profile Tensors of size > uint32 max. Got dim: ",
-        dim);
 
     tensor_metadata_.emplace_back(
         /*ptr_=*/(void*)t.unsafeGetTensorImpl(),
         /*dtype_=*/t.scalar_type(),
-        /*dim_=*/(uint32_t)dim,
-        /*layout_=*/t.layout());
-
-    for (const auto i : sizes) {
-      tensor_sizes_.emplace_back(i);
-    }
+        /*layout_=*/t.layout(),
+        /*sizes_and_strides_*/ t.unsafeGetTensorImpl()->sizes_and_strides());
```

Comment:
heads up, this would still error out for nested tensor (because `sizes_custom` errors). I'm adding a test for nested tensor profiling in #82854 so hopefully this should be visible.

Reply:
Should we just add an entry to the enum for "this doesn't even make sense"? Just want to future proof things a bit since I'm sure nested tensor won't be the last non-rectangular thing that people dream up.

Reply:
One could conceivably say that .sizes() of a nested tensor should return a vector of sizes of constituent tensors, or, if nested tensor happens to be rectangular, the sizes of this rectangular box, or something else? But until nested tensor decides what it wants profiler should ignore it. I don't know if we should call sizes_and_strides with try-catch to ignore all special tensorImpls that choose to throw an error for those calls, and not hardcode just nested tensors? E.g. sparse tensors will give you sizes, but will balk at strides, and I don't remember if their `strides_custom` throws an error.